how-to/hints-and-tips/docs/appassets-and-launch-external-process.md

```diff
@@ -0,0 +1,279 @@
+
+We have an article about the ability to set granular permissions for launching app assets so please refer to this article: [Launch Assets By Using A Secured API](https://developers.openfin.co/of-docs/docs/launch-assets-by-using-a-secured-api#launch-application-assets)
+
+Here is a snippet from that article that shows the permissions needed to be able to launch the app asset used in this example:
+
+```json
+"permissions": {
+    "System": {
+        "launchExternalProcess": {
+            "enabled": true,
+            "assets": {
+                "enabled": true,
+                "srcRules": [
+                    {
+                        "match": [
+                            "*://filesamples.com/*"
+                        ],
+                        "behavior": "allow"
+                    },
+                    {
+                        "match": [
+                            "<all_urls>"
+                        ],
+                        "behavior": "block"
+                    }
+                ]
+            }
+        }
+    }
+}
+```
+
+If you want to see where these permissions can go in the manifest please see our [protocol support](./protocol-support.md) article as it provides examples of setting permissions at the platform, window and view level.
+
+### Checking for permissions for App Assets and Launch External Process
+
+There is an api you can use to check to see if your application has been granted permission for a particular API.
+
+```js
+/**
+ * Do we have the permissions to launch external processes.
+ * @returns True if we have permission.
+ */
+async function getCanLaunchExternalProcess(): Promise<boolean> {
+ let canLaunchExternalProcess = false;
+
+ try {
+  const canLaunchExternalProcessResponse = await fin.System.queryPermissionForCurrentContext(
+   "System.launchExternalProcess"
+  );
+
+  canLaunchExternalProcess = canLaunchExternalProcessResponse?.granted;
+ } catch (error) {
+  console.error("Error while querying for System.launchExternalProcess permission", error);
+ }
+
+ return canLaunchExternalProcess;
+}
+
+/**
+ * Do we have the permissions to download app assets.
+ * @returns True if we have permission.
+ */
+async function getCanDownloadAppAssets(): Promise<boolean> {
+ let canDownloadAppAssets = false;
+
+ try {
+  const canDownloadAppAssetsResponse =
+   await fin.System.queryPermissionForCurrentContext("System.downloadAsset");
+  canDownloadAppAssets = canDownloadAppAssetsResponse?.granted;
+ } catch (error) {
+  console.error("Error while querying for System.downloadAsset permission", error);
+ }
+
+ return canDownloadAppAssets;
+}
+```
+
+## Examples
+
+We have aim to have examples that can be cloned and run locally and one example that covers the use case of a C# sidecar/helper app is required by an OpenFin application can be found here: [how to use a sidecar application - basic](https://github.com/built-on-openfin/csharp-starter/tree/main/how-to.v2/use-a-sidecar-app-basic).
```

Comment:
Doesn't make sense
`We have aim to have examples that can be cloned`

Reply:
Updated